### PR TITLE
Sonar lint

### PR DESCRIPTION
**Pull request by LendvaiDenes**

### CODE REVIEW

File: parser/Parser/CIFF.h [hunk 1]
```diff
@@ -71,7 +71,7 @@ class CIFF {
 		caption = r.readUntilChar('\n');
 
 		// -1 because "\n" is dropped from caption (readUntilChar). That one character is still part of the file, however.
-		uint64_t header_size_without_tags = sizeof(magic) + sizeof(header_size) + sizeof(content_size) + sizeof(width) + sizeof(height) + caption.size() + 1;
+		uint64_t header_size_without_tags = magic.size() + sizeof(header_size) + sizeof(content_size) + sizeof(width) + sizeof(height) + caption.size() + 1;
```
Comment (radl97) [inline parser/Parser/CIFF.h]:
```suggestion
		uint64_t header_size_without_tags = 4 + sizeof(header_size) + sizeof(content_size) + sizeof(width) + sizeof(height) + caption.size() + 1;
```

File: parser/Parser/CAFF_HEADER.h [hunk 2]
```diff
@@ -7,21 +7,21 @@ const uint8_t BLOCK_HEADER_ID = 0x1;
...
             throw ParsingException();
         }
-        r.readArray(magic, 4);
+        magic = r.readStringOfLength(4);
 		if (magic[0] != 'C' || magic[1] != 'A' || magic[2] != 'F' || magic[3] != 'F') {
 			throw ParsingException();
 		}
         r.readPrimitive(headerSize);
-        if (headerSize != sizeof(magic) + sizeof(headerSize) + sizeof(animationNumber)) {
+        if (headerSize != magic.size() + sizeof(headerSize) + sizeof(animationNumber)) {
```
Comment (radl97) [inline parser/Parser/CAFF_HEADER.h]:
```suggestion
        if (headerSize != 4 + sizeof(headerSize) + sizeof(animationNumber)) {
```

File: parser/Parser/CAFF_HEADER.h [hunk 3]
```diff
@@ -7,21 +7,21 @@ const uint8_t BLOCK_HEADER_ID = 0x1;
...
 
     void read(Reader& r) {
         Block::readBlockHeaderWithCheck(r, 0x1);
-        if (lengthOfBlock != sizeof(magic) + sizeof(headerSize) + sizeof(animationNumber)) {
+        if (lengthOfBlock != 4 + sizeof(headerSize) + sizeof(animationNumber)) {
             throw ParsingException();
         }
-        r.readArray(magic, 4);
+        magic = r.readStringOfLength(4);
 		if (magic[0] != 'C' || magic[1] != 'A' || magic[2] != 'F' || magic[3] != 'F') {
```
Comment (radl97) [inline parser/Parser/CAFF_HEADER.h]:
```suggestion
		if (magic != "CAFF") {
```

File: parser/Parser/CIFF.h [hunk 4]
```diff
@@ -51,7 +51,7 @@ class CIFF {
 public:
 
 	void read(Reader& r) {
-		r.readArray(magic, 4);
+		magic = r.readStringOfLength(4);
 		if (magic[0] != 'C' || magic[1] != 'I' || magic[2] != 'F' || magic[3] != 'F') {
```
Comment (radl97) [inline parser/Parser/CIFF.h]:
```suggestion
		if (magic != "CAFF") {
```

File: parser/Parser/CAFF.h [hunk 5]
```diff
@@ -8,9 +8,8 @@
 #include <vector>
 class CAFF
 {
-    bool isCreditPresent = false;
-	
 public:
+    bool isCreditPresent = false;
```
Comment (radl97) [inline parser/Parser/CAFF.h]: Can this be converted to a local?

**Review by radl97 — CHANGES_REQUESTED** [review]
I would request some small changes here and there.

There are two `magic.size()`, where I would use `4` or something that can be seen that it is constant (unlike `header_size`, `content_size`, `tags_size`, etc.)

File: parser/Parser/CAFF.h (hunk 5 above)
Comment (LendvaiDenes) [reply parser/Parser/CAFF.h]: fixed

File: parser/Parser/CAFF_HEADER.h (hunk 2 above)
Comment (LendvaiDenes) [reply parser/Parser/CAFF_HEADER.h]: fixed

File: parser/Parser/CAFF_HEADER.h (hunk 3 above)
Comment (LendvaiDenes) [reply parser/Parser/CAFF_HEADER.h]: fixed

File: parser/Parser/CIFF.h (hunk 1 above)
Comment (LendvaiDenes) [reply parser/Parser/CIFF.h]: fixed

**Review by radl97 — APPROVED** [review]

**Pull request merged**